### PR TITLE
Property test for `simplify` and `interpret`

### PR DESCRIPTION
**Pull request by curiousleo**
Just a few little QuickCheck property tests making use of the `instance Arbitrary Expr` from https://github.com/ucsd-progsys/liquid-fixpoint/pull/513.

```
  simplify does not increase expression size
    PLE:                          OK (0.03s)
      +++ OK, passed 500 tests:
      97.2% reduced size by 0                                                                                                                                            
       1.4% reduced size by 2                                                                                                                                            
       1.2% reduced size by 1                                                                                                                                            
       0.2% reduced size by 3                                                                                                                                            
    Interpreter:                  OK (0.02s)
      +++ OK, passed 500 tests:
      97.6% reduced size by 0                                                                                                                                            
       2.0% reduced size by 1                                                                                                                                            
       0.2% reduced size by 2                                                                                                                                            
       0.2% reduced size by 7                                                                                                                                                                
```

### CODE REVIEW

**Review by ranjitjhala — APPROVED** [review]

**Comment by ranjitjhala** [comment]
Thank you! Shall I merge?

**Comment by curiousleo** [comment]
> Thank you! Shall I merge?

Yes please! I have other ideas for property tests, but I think these changes stand on their own and are ready to merge.

**Comment by curiousleo** [comment]
I've added one more test establishing that `interpret` computes a fixpoint.

Good to merge in my opinion.

File: src/Language/Fixpoint/Solver/Interpreter.hs [hunk 1]
```diff
@@ -503,18 +511,20 @@ interpret ie γ ctx env   (ETApp e1 t)   = let e1' = interpret' ie γ ctx env e1
...
-    go [p] []     = interpret' ie γ ctx env p
-    go acc []     = PAnd acc
-    go acc (e:es) = if e == PTrue then go acc es
-                                  else if e == PFalse then PFalse else go (e:acc) es
+    go []  []         = PTrue
+    go [p] []         = interpret' ie γ ctx env p
+    go acc []         = PAnd acc
+    go acc (PTrue:es) = go acc es
+    go _   (PFalse:_) = PFalse
+    go acc (e:es)     = go (e:acc) es
```
Comment (curiousleo) [inline src/Language/Fixpoint/Solver/Interpreter.hs:519]: Purely a stylistic refactoring. I think this is easier to read, but I'm happy to revert this change if there is opposition.

**Review by facundominguez — APPROVED** [review]
LGTM!
